src/components/ProgressForm/InputWithQuestions.tsx

```diff
@@ -1,26 +1,55 @@
 import { inputPropsTypes } from '@/types/ProgressUpdates';
 import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+import { useRouter } from 'next/router';
 
 const InputWithQuestions = ({
     name,
     question,
     value,
     onChange,
 }: inputPropsTypes) => {
+    const { query } = useRouter();
+    const isDev = query.dev === 'true';
     return (
-        <div className={styles.inputComponent}>
-            <label className={styles.label} htmlFor={name} aria-label={name}>
-                {question}
-            </label>
-            <textarea
-                className={styles.input}
-                value={value}
-                onChange={(e) =>
-                    onChange({ type: name, value: e.target.value })
-                }
-                id={name}
-            />
-        </div>
+        <>
+            {isDev ? (
+                <div className={styles.inputComponentUpdated}>
+                    <label
+                        className={styles.labelUpdated}
+                        htmlFor={name}
+                        aria-label={name}
+                    >
+                        {question}
+                    </label>
+                    <textarea
+                        className={styles.inputUpdated}
+                        value={value}
+                        onChange={(e) =>
+                            onChange({ type: name, value: e.target.value })
+                        }
+                        id={name}
+                    />
+                </div>
+            ) : (
+                <div className={styles.inputComponent}>
+                    <label
+                        className={styles.label}
+                        htmlFor={name}
+                        aria-label={name}
+                    >
+                        {question}
+                    </label>
+                    <textarea
+                        className={styles.input}
+                        value={value}
+                        onChange={(e) =>
+                            onChange({ type: name, value: e.target.value })
+                        }
+                        id={name}
+                    />
+                </div>
+            )}
+        </>
```

Comment:
Can't we dynamically set the styles where dev is true instead of duplicating the entire component since only the styles have changed here?
Something like this `isDev ? styles.labelUpdated : styles.label`

Reply:
While using dynamic styles like `isDev ? styles.labelUpdated : styles.label` might seem cleaner initially, I chose to keep separate implementations for better maintainability and clarity. The current structure makes it easier to handle future changes or additions specific to each mode, makes debugging simpler, and provides clearer separation between the two variants. The minor duplication is a worthwhile trade-off for these benefits. If we later need to add more mode-specific features or behavior, this approach will scale better without requiring refactoring.